ga4gh/decorators.py

```diff
@@ -0,0 +1,92 @@
+"""
+General purpose decorators
+"""
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import functools
+import signal
+import time
+
+
+class TimeoutException(Exception):
+    """
+    A process has taken too long to execute
+    """
+
+
+class Repeat(object):
+    """
+    A decorator to use for repeating a tagged function.
```

Comment:
This is the sort of might-as-well-be-code that I don't like to put in decorators, but I don't have a well-formed objection. Can you give an example of how it'd be useful? (github code search isn't finding it for me, if there is one already...)

I've skimmed PEP318, and the examples I see there are the more meta things I'd expected - adding typing, metadata, and perhaps patterns. Aspect-orientation also makes sense.